cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/existingdataproc/ExistingDataprocProvisioner.java

```diff
@@ -0,0 +1,295 @@
+                                           + " . Validate the cluster exists and " +
+                                           "service account has access to the cluster");
+    } catch (GeneralSecurityException e) {
+      throw new IllegalArgumentException("Security Exception Accessing Cluster " + clusterName +
+                                           " . Validate the cluster exists and service " +
+                                           "account has access to the cluster");
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Error Connecting to  Cluster " + clusterName +
+                                           " . Validate the cluster exists and service " +
+                                           "account has access to the cluster");
+    }
+  }
+
+  /**
+   * This is similar to the Remote Hadoop provisioner.
+   * The cluster will be validated again to ensure it still exists before start the pipeline.
+   *
+   * @param context provisioner context
+   * @return
+   * @throws Exception
+   */
+  @Override
+  public Cluster createCluster(ProvisionerContext context) throws Exception {
+    // Reload system context properties and get system labels
+    systemContext.reloadProperties();
+    ExistingDataprocConf conf = ExistingDataprocConf.create(createContextProperties(context));
+
+    if (conf.getKeyPair() != null) {
+      context.getSSHContext().setSSHKeyPair(conf.getKeyPair());
+    }
+
+    String clusterName = conf.getClusterName();
+    try (DataprocClient client =
+           DataprocClient.fromConf(conf)) {
+      Optional<Cluster> existing = client.getCluster(clusterName);
+      if (existing.isPresent()) {
+        //Set CDAP Version label for tracking
+        ClusterOperationMetadata createOperationMeta = client.setSystemLabels(getSystemLabels(systemContext),
+                                                                              clusterName);
+        int numWarnings = createOperationMeta.getWarningsCount();
+        if (numWarnings > 0) {
+          LOG.warn("Encountered {} warning{} while creating Dataproc cluster:\n{}",
+                   numWarnings, numWarnings > 1 ? "s" : "",
+                   String.join("\n", createOperationMeta.getWarningsList()));
+        }
+      }
+      return existing.get();
+    }
+  }
+
+  @Override
+  public ClusterStatus getClusterStatus(ProvisionerContext context, Cluster cluster) throws Exception {
+    ClusterStatus status = cluster.getStatus();
+    return status == ClusterStatus.DELETING ? ClusterStatus.NOT_EXISTS : status;
+  }
+
+  @Override
+  public Cluster getClusterDetail(ProvisionerContext context, Cluster cluster) throws Exception {
+    return new Cluster(cluster, getClusterStatus(context, cluster));
+  }
+
+  @Override
+  public void deleteCluster(ProvisionerContext context, Cluster cluster) throws Exception {
+    //Need to evaluate if any job clean up needs to be done.
```

Comment:
You have to delete files on GCS for the run as indicated by the context and cluster.